iot_config_client/src/iot_config_client.rs

```diff
@@ -0,0 +1,114 @@
+type IotAdminConfigClient = iot_config::admin_client::AdminClient<Channel>;
+
+#[derive(Debug, Clone)]
+pub struct IotConfigClient {
+    keypair: Arc<Keypair>,
+    pub gateway_client: IotGatewayConfigClient,
+    pub region_client: IotAdminConfigClient,
+    batch_size: u32,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum IotConfigClientError {
+    #[error("error resolving gateway info")]
+    GatewayResolverError(#[from] gateway_resolver::GatewayResolverError),
+    #[error("error resolving region params")]
+    RegionParamsResolverError(#[from] region_params_resolver::RegionParamsResolverError),
+    #[error("gateway not found: {0}")]
+    GatewayNotFound(PublicKeyBinary),
+    #[error("region params not found {0}")]
+    RegionParamsNotFound(String),
+    #[error("uri error")]
+    Uri(#[from] http::uri::InvalidUri),
+    #[error("grpc {}", .0.message())]
+    Grpc(#[from] tonic::Status),
+    #[error("keypair error: {0}")]
+    KeypairError(#[from] Box<helium_crypto::Error>),
+    #[error("signing error: {0}")]
+    SignError(#[from] helium_crypto::Error),
+}
+
+#[async_trait::async_trait]
+impl GatewayInfoResolver for IotConfigClient {
+    async fn resolve_gateway_info(
+        &mut self,
+        address: &PublicKeyBinary,
+    ) -> Result<GatewayInfo, IotConfigClientError> {
+        let mut req = GatewayInfoReqV1 {
+            address: address.clone().into(),
+            signature: vec![],
+        };
+        req.signature = self.keypair.sign(&req.encode_to_vec())?;
+        let res = self.gateway_client.info(req).await?.into_inner();
+        match res.info {
+            Some(gateway_info) => Ok(gateway_info.try_into()?),
+            _ => Err(IotConfigClientError::GatewayNotFound(address.clone())),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl RegionParamsResolver for IotConfigClient {
+    async fn resolve_region_params(
+        &mut self,
+        region: Region,
+    ) -> Result<RegionParamsInfo, IotConfigClientError> {
+        let mut req = RegionParamsReqV1 {
+            region: region.into(),
+            signature: vec![],
+        };
+        req.signature = self.keypair.sign(&req.encode_to_vec())?;
+        match self.region_client.region_params(req).await {
+            Ok(region_params_info) => Ok(region_params_info.into_inner().try_into()?),
+            _ => Err(IotConfigClientError::RegionParamsNotFound(format!(
+                "{region:?}"
+            ))),
+        }
+    }
+}
+
+impl IotConfigClient {
+    pub fn from_settings(settings: &Settings) -> Result<Self, IotConfigClientError> {
+        let keypair = settings.keypair()?;
+        Ok(Self {
+            keypair: Arc::new(keypair),
+            gateway_client: settings.connect_iot_gateway_config(),
+            region_client: settings.connect_iot_region_config(),
+            batch_size: settings.batch,
+        })
+    }
+
+    pub async fn active_gateways(&mut self) -> Result<GatewayInfoStream, IotConfigClientError> {
```

Comment:
can we rename this function to something like `info_stream` or `gateway_stream`? it used to return `active_gateways` when we wrote the original stream coming from the blockchain-node since we could do the "active" filtering from the blockchain side but since we can't do that now and we only get the "active" status after comparing to the records in the verifier's DB this name seems like it will just add confusion anymore

Reply:
changed here: 7f7555aab495ad2f224496c8994d4058104482b6